server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -130,20 +134,122 @@ public void testNodesJoinAfterStableCluster() {
         cluster.stabilise();
 
         final long currentTerm = cluster.getAnyLeader().coordinator.getCurrentTerm();
-        final int newNodesCount = randomIntBetween(1, 2);
-        cluster.addNodes(newNodesCount);
-        cluster.stabilise(
-            // The first pinging discovers the master
-            defaultMillis(DISCOVERY_FIND_PEERS_INTERVAL_SETTING)
-                // One message delay to send a join
-                + DEFAULT_DELAY_VARIABILITY
-                // Commit a new cluster state with the new node(s). Might be split into multiple commits
-                + newNodesCount * DEFAULT_CLUSTER_STATE_UPDATE_DELAY);
+        cluster.addNodesAndStabilise(randomIntBetween(1, 2));
 
         final long newTerm = cluster.getAnyLeader().coordinator.getCurrentTerm();
         assertEquals(currentTerm, newTerm);
     }
 
+    public void testExpandsConfigurationWhenNodesJoinAndContractsWhenTheyLeave() {
```

Comment:
I think we can have separate tests here:
- 1 to 3
- 3 to 5
- 5 to 3
- 3 to 1

Reply:
Ok, I added finer-grained tests in 5692417e2e6965df65dad6fbb694b5398c01d620. We cannot test 3-to-1 without retirement, but there is a case for 3-to-2.